ee/server/service/vpp.go

```diff
@@ -451,6 +451,88 @@ func getVPPAppsMetadata(ctx context.Context, ids []fleet.VPPAppTeam) ([]*fleet.V
 	return apps, nil
 }
 
+func (svc *Service) UpdateAppStoreApp(ctx context.Context, titleID uint, teamID *uint, selfService bool, labelsIncludeAny, labelsExcludeAny []string) (*fleet.VPPAppStoreApp, error) {
+	if err := svc.authz.Authorize(ctx, &fleet.VPPApp{TeamID: teamID}, fleet.ActionWrite); err != nil {
+		return nil, err
+	}
+
+	var teamName string
+	if teamID != nil && *teamID != 0 {
+		tm, err := svc.ds.Team(ctx, *teamID)
+		if fleet.IsNotFound(err) {
+			return nil, fleet.NewInvalidArgumentError("team_id", fmt.Sprintf("team %d does not exist", *teamID)).
+				WithStatus(http.StatusNotFound)
+		} else if err != nil {
+			return nil, ctxerr.Wrap(ctx, err, "UpdateAppStoreApp: checking if team exists")
+		}
+
+		teamName = tm.Name
+	}
+
+	validatedLabels, err := ValidateSoftwareLabels(ctx, svc, labelsIncludeAny, labelsExcludeAny)
+	if err != nil {
+		return nil, ctxerr.Wrap(ctx, err, "UpdateAppStoreApp: validating software labels")
+	}
+
+	meta, err := svc.ds.GetVPPAppMetadataByTeamAndTitleID(ctx, teamID, titleID)
+	if err != nil {
+		return nil, ctxerr.Wrap(ctx, err, "UpdateAppStoreApp: getting vpp app metadata")
+	}
+
+	if selfService && meta.Platform != fleet.MacOSPlatform {
+		return nil, fleet.NewUserMessageError(errors.New("Currently, self-service only supports macOS"), http.StatusBadRequest)
+	}
+
+	appToWrite := &fleet.VPPApp{
+		VPPAppTeam: fleet.VPPAppTeam{
+			VPPAppID: fleet.VPPAppID{
+				AdamID: meta.AdamID, Platform: meta.Platform,
+			},
+			SelfService: selfService,
+		},
+		TeamID:           teamID,
+		TitleID:          titleID,
+		ValidatedLabels:  validatedLabels,
```

Comment:
This should be at the VPPAppTeam level rather than the VPPApp level right?

Reply:
Fair point @iansltx , will update